### ISSUE TITLE
Extend Offload Copy API with additional arguments for dcp

### ISSUE THREAD

**Issue by mcfadden8** [NearNodeFlash/nnf-dm#97]
The underlying dcp command takes many useful options including an option to follow symbolic links and adjust the block and chunk size used for moving data.  This request is for the client side API to be extended to take an optional string containing additional command line arguments to be passed along to the dcp command.

Is this possible?

**Comment by bdevcich-hpe** [comment]
Yes, this is possible and it's something I've touched on with #90. On the directive side of things (via `copy_in`/`copy_out`) we've allowed for configuration to completely override the command that is performing data movement:

```
command: mpirun --allow-run-as-root --hostfile $HOSTFILE dcp --progress 1 --uid $UID --gid $GID $SRC $DEST
```

This allows for a full override of the data movement command including mpirun and dcp options. We could do something similar for the copy offload side of things if there is a need to change these values on a per request basis.

Is there ever a need to adjust `mpirun` from the API? Or just `dcp`?

**Comment by mcfadden8** [comment]
I think that there may be times when we want to have dcp follow symbolic links and we may want to "play" with the chunk and block size that is used by the copy depending upon the size of the files and the number of them to be copied etc.

I'm not sure about the mpi adjustments, but I guess that might also be a possibility.

**Comment by bdevcich-hpe** [comment]
We can add extra `dcp` options to the Copy Offload API. I will just have to think about how that fits in with the configuration I mentioned in my earlier comment.

Perhaps we add something like `$DCP_OPTS` that can be overridden by the Copy Offload API:

```
command: mpirun --allow-run-as-root --hostfile $HOSTFILE dcp --progress 1 --uid $UID --gid $GID $DCP_OPTS $SRC $DEST
```

For example, if a `CreateRequest` supplies extra `dcp` options, then it would override that in the full command.

**Comment by bdevcich-hpe** [comment]
#99
